File: src/models/MarketplaceRequest.ts
import type { MarketplaceRequestStatus } from "@/types/auth";

export type MarketplaceRequestRow = {
  id: string;
  customer_id: string;
  project_title: string;
  description: string;
  service_type: string;
  material_type: string;
  location: string;
  deadline: string;
  drawing_urls: string[];
  technical_requirements: string | null;
  quantity: string | null;
  quality_expectations: string | null;
  budget_range: string | null;
  inspection_requirement: string | null;
  gst_requirement: boolean;
  status: MarketplaceRequestStatus;
  created_at: string;
};

export type MarketplaceRequest = {
  id: string;
  customerId: string;
  projectTitle: string;
  description: string;
  serviceType: string;
  materialType: string;
  location: string;
  deadline: string;
  drawingUrls: string[];
  technicalRequirements: string | null;
  quantity: string | null;
  qualityExpectations: string | null;
  budgetRange: string | null;
  inspectionRequirement: string | null;
  gstRequirement: boolean;
  status: MarketplaceRequestStatus;
  createdAt: string;
};

File: rkispro-android/app/src/main/java/com/rkispro/marketplace/data/model/MarketplaceRequest.kt
package com.rkispro.marketplace.data.model

import com.google.gson.annotations.SerializedName

data class MarketplaceRequest(
    val id: String,
    @SerializedName("customer_id") val customerId: String,
    @SerializedName("project_title") val projectTitle: String,
    val description: String,
    @SerializedName("service_type") val serviceType: String,
    @SerializedName("material_type") val materialType: String,
    val location: String,
    val deadline: String,
    @SerializedName("drawing_urls") val drawingUrls: List<String> = emptyList(),
    @SerializedName("technical_requirements") val technicalRequirements: String? = null,
    val quantity: String? = null,
    @SerializedName("quality_expectations") val qualityExpectations: String? = null,
    @SerializedName("budget_range") val budgetRange: String? = null,
    @SerializedName("inspection_requirement") val inspectionRequirement: String? = null,
    @SerializedName("gst_requirement") val gstRequirement: Boolean = false,
    val status: String = "Pending",
    @SerializedName("created_at") val createdAt: String? = null
)
